llvm/include/llvm/Transforms/Utils/LoopMetadata.h

```diff
@@ -31,6 +31,7 @@ class LoopMetadata : public PassInfoMixin<LoopMetadata> {
   ICmpInst *LoopCmpInstr;
   bool IsLoopIncrementing;
   int64_t LoopStepSize;
+  bool IsTruncatedSCEV;
```

Comment:
That seems dangerous to store such "volatile" state as pass state. COuld you find a way to reduce the scope of that variable?

Reply:
wouldn't this count for every loop dependent  member variables?

Reply:
Yes I think so :) But I'm biased, I hate state variables, as I find it makes it harder to understand what functions do